src/tracers.jl

```diff
@@ -117,30 +117,36 @@ $(TYPEDEF)
 
 For a higher-level interface, refer to [`hessian_pattern`](@ref).
 
+The last type parameter `shared` is a `Bool` indicating whether the Hessian should be shared among all intermediate scalar quantities.
```

Comment:
Could you emphasize that only the `hessian` field in `HessianTracer` is shared? 
The word "Hessian" is a bit ambiguous.

Reply:
I think that should be enough documentation until we decide to export shared pattern functionality.

It's going to be non-trivial to explain to users why they can obtain four different types of sparsity patterns.

Reply:
The same detail should be explained in the `create_tracers` docstring as well.

Reply:
I added a warning saying that `shared` is not part of the public API and should always be `false`. I'm not entirely happy with that because it means that users _do_ have to set it. But I think this is better than needing overloads for a partially-defined type `HessianTracer{G,H}`

Reply:
Yeah, I prefer enforcing full type definitions as well.